Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -402,6 +406,28 @@ TextureCache::TCacheEntryBase* TextureCache::Load(const u32 stage)
 		++iter;
 	}
 
+	// Allow to find this texture at a different address, if the texture was fully hashed and all other parameters match
+	// Tales of Symphonia (GC) uses over 500 small textures in menus, but only around 70 different ones
+	if (g_ActiveConfig.iSafeTextureCache_ColorSamples == 0 ||
+		std::max(texture_size, palette_size) <= (u32)g_ActiveConfig.iSafeTextureCache_ColorSamples * 8)
+	{
+		iter = textures.begin();
+
+		while (iter != textures.end())
```

Comment:
As you don't modify "iter", there is no need to write this loop explicit:
for (const auto& pair : textures )
{
TCacheEntryBase* entry = pair.second;
// no iter++
}